### ISSUE TITLE
Navigating to a navigator whose landing screen is a child navigator overrides the child params with the parent params

### ISSUE THREAD

**Issue by conleymon** [react-navigation/react-navigation#6850]
**Current Behavior**
const childNavigator= createStackNavigator({
   childScreen1: {
      screen: SomeScreen1,
      params: { paramName:'shouldBeChild'
   },
   childScreen2: {
      screen: SomeScreen2,
   }
})

const parentNavigator = createStackNavigator({
   parentScreen1:{
      screen: childNavigator,
      params:{
         paramName:'ButIsActuallyParent',
      }
})

NavigationService.navigate('ParentScreen1')
//submits 'ButIsActuallyParent' to the child screen upon instantiation

**Your Environment**
expo SDK 36 (managed)
react-native .61
latest react-navigation as of 2/13/2020
node 12.14.1
| software                       | version |
| ------------------------------ | ------- |
| BOTH iOS and Android                 | 
| @react-navigation/native       | latest as of 2/13/2020
| @react-navigation/native-stack | 
| react-native-screens           |
| react-native                   | .61
| expo                           | SDK 36
| node                           | 12.14.1
| yarn                    |

**Comment by sudomann** [comment]
I'm experiencing this as well. I was pulling my hair out (I'm currently bald) trying to set `gestureEnabled: false` on a stack navigator and it was being ignored. Now I know this is the cause. I verified this by setting that option in a parent navigator to get the effect I wanted, but then it affected other screens I did not need it enabled on.

**Comment by WoLewicki** [comment]
Could you provide a snack that would show the bug?

**Comment by conleymon** [comment]
Sure,
https://snack.expo.io/@datconley/73f08f


    On Friday, February 14, 2020, 04:33:50 AM CST, Wojciech Lewicki <notifications@github.com> wrote:  
 
 
Could you provide a snack that would show the bug?

**Comment by WoLewicki** [comment]
I think the problem might be located here: https://github.com/react-navigation/react-navigation/blob/1a28c299b5e3f0805eb6e9ea3cf5e9cc90c7a280/packages/core/src/useNavigationBuilder.tsx#L205 where parent params are added after the child ones. Changing it could be considered a breaking change. Wdyt @satya164?

**Comment by satya164** [comment]
That's merging passed params with intial params which is expected behavior. This report is for v4

**Comment by conleymon** [comment]
Is there any way to protect hard-coded params from parental override?

**Comment by WoLewicki** [comment]
It looks like there is no option for now since they are just overridden, but can't you use different param names in the child component?

**Comment by conleymon** [comment]
Hi, WoLewicki, Thanks for your attention to this. I was trying to stick to a paradigm. I needed to store certain component refs in state to access them from elsewhere in the program, I Wrote a little HOC that would look for a 'stateRef' property in the params and assign the ref of that component in global state. I did find a work around though(check a registry of special cases in the HOC). Thanks again
